code/modules/mob/living/carbon/xenomorph/evolution.dm

```diff
@@ -276,9 +276,15 @@
 		if(death_timer)
 			to_chat(src, span_warning("The hivemind is still recovering from the last [initial(new_caste_type.display_name)]'s death. We must wait [DisplayTimeText(timeleft(death_timer))] before we can evolve."))
 			return FALSE
-	var/maximum_active_caste = new_caste_type.maximum_active_caste
-	if(maximum_active_caste != INFINITY && maximum_active_caste <= length(hive.xenos_by_typepath[new_mob_type]))
-		to_chat(src, span_warning("There is already a [initial(new_caste_type.display_name)] in the hive. We must wait for it to die."))
+	var/maximum_active = new_caste_type.maximum_active
+	if(maximum_active != -1 && maximum_active <= length(hive.xenos_by_typepath[new_mob_type]))
```

Comment:
if it's INFINITY you don't need to check maxium_active != -1 or maxium_active != INFINITY

Reply:
If you want me to instead have to pass another value to the UI because trusting defines to stay the same is bad, be my guest.